envoy/common/io/io_uring.h

```diff
@@ -133,6 +159,109 @@ class IoUring {
 };
 
 using IoUringPtr = std::unique_ptr<IoUring>;
+class IoUringWorker;
+
+/**
+ * Abstract for each socket.
+ */
+class IoUringSocket {
+public:
+  virtual ~IoUringSocket() = default;
+
+  /**
+   * Get the IoUringWorker this socket bind to.
+   */
+  virtual IoUringWorker& getIoUringWorker() const PURE;
+
+  /**
+   * Return the raw fd.
+   */
+  virtual os_fd_t fd() const PURE;
+
+  /**
+   * On accept request completed.
+   * TODO (soulxu): wrap the raw result into a type. It can be `IoCallUint64Result`.
+   * @param req the AcceptRequest object which is as request user data.
+   * @param result the result of operation in the request.
+   * @param injected indicates the completion is injected or not.
+   */
+  virtual void onAccept(Request* req, int32_t result, bool injected) PURE;
+
+  /**
+   * On connect request completed.
+   * TODO (soulxu): wrap the raw result into a type. It can be `IoCallUint64Result`.
+   * @param req the request object which is as request user data.
```

Comment:
Request object is the user data we provide to each iouring request which is used to hold the request related data. Like for read/write, it will carry the buffer, for accept, it will carry the peer address.

https://github.com/envoyproxy/envoy/pull/24958/files#diff-df9a08b7c36ffbd691c64a9c0c22a34cd2848af1ce434bc05bd661dbba4b7199R41